qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -187,8 +189,8 @@ def _generate_circuit(
                 group_elt_gate = group_elt_circ
                 group_elt_op = Clifford(group_elt_circ)
 
-            if hasattr(group_elt_gate, "to_gate"):
-                group_elt_gate = group_elt_gate.to_gate()
+            if not isinstance(group_elt_gate, Instruction):
+                group_elt_gate = group_elt_gate.to_instruction()
```

Comment:
I think this has been already resolved with https://github.com/Qiskit/qiskit-experiments/pull/762. Currently test is missing because support of Delay requires the latest terra which was unstable at the time I merged the PR. What is the point using `to_instruction`?

Reply:
The fix in #762 does not cover the case of interleaving a circuit with delays, which is tested in the test case below. If we interleave a QuantumCircuit object with delay, it does not have `to_gate` method, and it fails to be appended to a circuit afterwards.

Reply:
Make sense. I've forgotten the interleaved element typehint says `QuantumCircuit` is acceptable.